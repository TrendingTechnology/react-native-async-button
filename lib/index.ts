import ContextAsyncButton from '@lib/ContextAsyncButton';
export { ContextAsyncButton };

import AsyncButton from '@lib/AsyncButton';
export * from '@lib/AsyncButton';
export { AsyncButton };
export default AsyncButton;
