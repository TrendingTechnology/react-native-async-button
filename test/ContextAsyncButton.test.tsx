import { } from 'jest';

import * as React from 'react';
import { Text } from 'react-native';
import { ContextAsyncButton } from '../lib';

import * as renderer from 'react-test-renderer';

interface IContext {
  data: number;
}

async function success(_: IContext): Promise<void> {
  await new Promise<void>(resolve => setTimeout(resolve, 0));
}

async function failure(context: IContext): Promise<void> {
  await success(context);
  throw new Error();
}

const idleComponent = () => <Text>Hey, there!</Text>;

// TODO: https://github.com/Microsoft/TypeScript/issues/6395
interface ISpecializedContextAsyncButton { new(): ContextAsyncButton<IContext>; }
// tslint:disable-next-line:variable-name
const SpecializedContextAsyncButton = ContextAsyncButton as ISpecializedContextAsyncButton;

describe('ContextAsyncButton', () => {
  describe('successful asynchronous operation', () => {
    it('should render', () => {
      const context: IContext = { data: 1 };
      const tree = renderer.create(
        <SpecializedContextAsyncButton context={context} onPress={success} IdleComponent={idleComponent} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('failing asynchronous operation', () => {
    it('should render', () => {
      const context: IContext = { data: 1 };
      const tree = renderer.create(
        <SpecializedContextAsyncButton context={context} onPress={failure} IdleComponent={idleComponent} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
