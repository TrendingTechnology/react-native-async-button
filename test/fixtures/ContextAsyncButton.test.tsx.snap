// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextAsyncButton failing asynchronous operation should render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "minHeight": 44,
        "minWidth": 44,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    Hey, there!
  </Text>
</View>
`;

exports[`ContextAsyncButton successful asynchronous operation should render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "minHeight": 44,
        "minWidth": 44,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    Hey, there!
  </Text>
</View>
`;
